src/main/java/com/enonic/lib/cron/scheduler/JobExecutionCommand.java

```diff
@@ -0,0 +1,81 @@
+package com.enonic.lib.cron.scheduler;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Stopwatch;
+
+import com.enonic.lib.cron.model.JobDescriptor;
+
+final class JobExecutionCommand
+    implements Runnable
+{
+    private final static Logger LOG = LoggerFactory.getLogger( JobExecutionCommand.class );
+
+    private final JobDescriptor descriptor;
+
+    private final AtomicInteger runCount;
+
+    private final Consumer<JobExecutionCommand> runCallback;
+
+    private final Predicate<JobDescriptor> runCheckFunction;
+
+    JobExecutionCommand( final JobDescriptor descriptor, final Consumer<JobExecutionCommand> runCallback )
+    {
+        this.descriptor = descriptor;
+        this.runCallback = runCallback;
+        this.runCount = new AtomicInteger( 0 );
+        this.runCheckFunction = ( JobDescriptor o ) -> o.getTimes().isEmpty() || this.runCount.get() < o.getTimes().get();
+    }
+
+    @Override
+    public void run()
+    {
+        try
+        {
+            if ( !runCheckFunction.test( this.descriptor ) )
+            {
+                this.runCallback.accept( this );
+                return;
+            }
+
+            this.doRun();
+            this.runCount.incrementAndGet();
```

Comment:
`getAndIncrement` could be used instead straight in `runCheckFunction`

Reply:
there is also usage of `runCheckFunction` in `isFinished`

Reply:
lets fix in #15